module-system/sov-cli/src/wallet_state.rs

```diff
@@ -52,6 +53,72 @@ impl<Tx: Serialize + DeserializeOwned, Ctx: sov_modules_api::Context> WalletStat
     }
 }
 
+/// A struct representing private key and associated address
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(bound = "Ctx::Address: Serialize + DeserializeOwned")]
+pub struct PrivateKeyAndAddress<Ctx: sov_modules_api::Context> {
+    /// Private key of the address
+    pub private_key: Ctx::PrivateKey,
+    /// Address associated from the private key
+    pub address: Ctx::Address,
+}
+
+impl<Ctx: sov_modules_api::Context> PartialEq for PrivateKeyAndAddress<Ctx> {
+    fn eq(&self, other: &Self) -> bool {
+        self.private_key.pub_key() == other.private_key.pub_key() && self.address == other.address
+    }
+}
+
+impl<Ctx: sov_modules_api::Context> PrivateKeyAndAddress<Ctx> {
+    /// Returns boolean if address actually matches the private key
+    pub fn is_matching(&self) -> bool {
+        self.private_key.to_address::<Ctx::Address>() == self.address
+    }
+
+    /// Randomly generates a new private key and address
+    pub fn generate() -> Self {
+        let private_key = Ctx::PrivateKey::generate();
+        let address = private_key.to_address::<Ctx::Address>();
+        Self {
+            private_key,
+            address,
+        }
+    }
+
+    /// Generates valid private key and address from given private key
+    pub fn from_key(private_key: Ctx::PrivateKey) -> Self {
+        let address = private_key.to_address::<Ctx::Address>();
+        Self {
+            private_key,
+            address,
+        }
+    }
+}
+
+/// A simplified struct representing private key and associated address
+/// where the private key is represented as a hex string
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct HexPrivateAndAddress {
+    /// Private key is hex encoded bytes, without leading 0x
+    pub hex_priv_key: String,
+    /// Address is in canonical string format
+    pub address: String,
+}
+
+impl<Ctx: sov_modules_api::Context> TryFrom<HexPrivateAndAddress> for PrivateKeyAndAddress<Ctx> {
```

Comment:
Should we add a conversion going the other way (`From<PrivateKeyAndAddress<Ctx>> for HexPrivateAndAddress`)?

Reply:
Yes, that would be super simple. I didn't add it, because there was no use case. But those types are interchangable.

Reply:
So this actually wouldn't work without updating `PrivateKey` trait. Currently it does not have `ToBytes` or `AsRef<[u8]>` or something that will allow to get bytes and encode them.

Reply:
Plan is to remove HexPrivateKey: https://github.com/Sovereign-Labs/sovereign-sdk/issues/766